clang/docs/LanguageExtensions.rst

```diff
@@ -2311,6 +2311,30 @@ argument.
   int *pb =__builtin_preserve_access_index(&v->c[3].b);
   __builtin_preserve_access_index(v->j);
 
+``__builtin_unique_stable_name``
+--------------------------------
+
+``__builtin_unique_stable_name()`` is a builtin that takes a type or unevaluated
+expression and produces a string literal containing a unique name for the type
+(or type of the expression) that is stable across split compilations.
+
+In cases where the split compilation needs to share a unique token for a type
+across the boundary (such as in an offloading situation), this name can be used
+for lookup purposes.
+
+This builtin is superior to RTTI for this purpose for two reasons.  First, this
+value is computed entirely at compile time, so it can be used in constant
+expressions. Second, this value encodes lambda functions based on line-number
+rather than the order in which it appears in a function. This is valuable
+because it is stable in cases where an unrelated lambda is introduced
+conditionally in the same function.
+
+The current implementation of this builtin uses a slightly modified Itanium
+Mangler to produce the unique name. The lambda ordinal is replaced with one or
+more line/column pairs in the format ``LINE->COL``, separated with a ``~``
```

Comment:
@erichkeane - Just noticed that this is the description for the old mangling format. Could you please update it?
Commenting here in case it escaped your attention too.

Reply:
I'd forgotten about that, thanks.  I'm doing a bunch of other stuff on this patch as well right now, but it would probably be a good learning opportunity for you to write this up so we can review/fill in the gaps.  Please submit a PR when you get a chance.  Tests are more important obviously, but this needs to happen next week as well.